File: backend/domain/entities/mensagem_contato.py
"""Entidade de domínio: MensagemContato."""

import uuid
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class MensagemContato:
    nome: str
    email: str
    mensagem: str
    id: uuid.UUID = field(default_factory=uuid.uuid4)
    telefone: str | None = None
    lida: bool = False
    criada_em: datetime = field(default_factory=datetime.utcnow)

File: frontend/core/domain/entities/MensagemContato.ts
export interface PlainMensagemContato {
  id: string;
  nome: string;
  email: string;
  telefone: string | null;
  mensagem: string;
  lida: boolean;
  criadaEm: string;
}

export class MensagemContato {
  constructor(
    public readonly id: string,
    public readonly nome: string,
    public readonly email: string,
    public readonly telefone: string | null,
    public readonly mensagem: string,
    public readonly lida: boolean,
    public readonly criadaEm: string
  ) {}
}
